frontend/server/fileSystem.js

```diff
@@ -4,6 +4,42 @@ import fs from "fs/promises";
 import { tmpdir } from "os";
 import path from "path";
 
+//fix for specs.json issue
+const coreBlockDir = 'core/blocks';
+
+// Function to rename specs_v1.json to specs.json if specs.json doesn't exist
+export const renameSpecsIfNotExists = async (dir) => {
+  const specsV1Path = path.join(dir, 'specs_v1.json');
+  const specsPath = path.join(dir, 'specs.json');
+  
+  if (await fileExists(specsV1Path) && !(await fileExists(specsPath))) {
+    await fs.rename(specsV1Path, specsPath);
+    console.log(`Renamed ${specsV1Path} to ${specsPath}`);
+  } else {
+    console.log(`specs.json already exists in ${dir} or specs_v1.json does not exist`);
+  }
+}
+
+// Function to process all processing blocks in the core block directory
+const processCoreBlockDirectory = async (coreBlockDir) => {
+  try {
+    const processingBlocks = await fs.readdir(coreBlockDir);
+    for (const block of processingBlocks) {
+      const blockPath = path.join(coreBlockDir, block);
+      const stat = await fs.stat(blockPath);
+      if (stat.isDirectory()) {
+        await renameSpecsIfNotExists(blockPath);
+      }
+    }
+  } catch (error) {
+    console.error("Error processing core block directory:", error);
+  }
+}
+
+// Call function to process core block directory
+processCoreBlockDirectory(coreBlockDir);
+
+
```

Comment:
Is this supposed to be here or in an other MR?

Reply:
Yes it is, I'll fix it in the next PR, this one had a lot of conflicts, and issues